segmentstore/server/src/main/java/io/pravega/segmentstore/server/WriterFactory.java

```diff
@@ -25,8 +26,21 @@
      * @param attributeIndex    The ContainerAttributeIndex to attach to (to durably store Extended Attributes for
      *                          processed appends).
      * @param storage           The Storage adapter to use.
+     * @param createProcessors  A Function that, when invoked with a Segment's Metadata, will create any additional
+     *                          WriterSegmentProcessors for that Segment.
      * @return An instance of a class that implements the Writer interface.
      */
     Writer createWriter(UpdateableContainerMetadata containerMetadata, OperationLog operationLog, ReadIndex readIndex,
-                        ContainerAttributeIndex attributeIndex, Storage storage);
+                        ContainerAttributeIndex attributeIndex, Storage storage, CreateProcessors createProcessors);
+
+    @FunctionalInterface
+    interface CreateProcessors {
```

Comment:
I'm curious to know how we will be adding extensions that we develop over time. Is there going to be an implementation of this interface somewhere that we add to?

Reply:
This is a functional interface. It defines a lambda function signature that can be passed in. This is internal; you don't need anything extra. 

The way you add additional extensions is by registering them in `ServiceBuilder`. You will see in a subsequent PR which depends on this one.